### PR TITLE
Avoid redudant requests in decipher process

### PR DESCRIPTION
**Pull request by Julian-Chu**
# Description
* move decipher from parseVideoInfo into StartDownload method
* merge  FormatBase and stream into Stream type
* remove log to show stream info in parseVideoInfo
* use argument -info instead of -itags

## Issues to fix

Please link issues this PR will fix: \
\#67

if no relevant issue, but this will fix something important for reference
, please free to open an issue.

### CODE REVIEW

File: youtube.go [hunk 1]
```diff
@@ -432,7 +413,9 @@ func (y *Youtube) getVideoInfo() error {
 	if err != nil {
 		return err
 	}
-	defer resp.Body.Close()
+	defer func() {
+		err = resp.Body.Close()
```
Comment (Julian-Chu) [inline youtube.go:417]: mistake,  if err not nil, here will overwrite existing err

Comment (kkc) [reply youtube.go:417]: I was wondering if we can adopt Thanos [runutil](https://thanos.io/contributing/coding-style-guide.md/#exhaust-readers) to handle this kind of error

Comment (Julian-Chu) [reply youtube.go]: We can,  but  I found ExhaustCloseWithLogOnErr can be implemented without this pkg, so I write a custom close method.

**Comment by kkc** [comment]
LGTM

**Pull request merged**